api/candlepin-api-spec.yaml

```diff
@@ -1124,6 +1124,390 @@ paths:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Either role or username is null or empty
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        404:
+          description: Role or Username not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role or Username not found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    delete:
+      description: Removes a User from a Role
+      tags:
+        - Roles
+      operationId: deleteUserFromRole
+      parameters:
+        - in: path
+          name: role_name
+          description: Role name
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: username
+          description: User name
+          required: true
+          schema:
+            type: string
+      security: []
+      responses:
+        200:
+          description: User from a role is removed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoleDTO'
+        400:
+          description: Either role or username is null or empty
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Either role or username is null or empty
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        404:
+          description: Role or Username not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Role or Username not found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+  /roles/{role_name}/permissions:
+    post:
+      tags:
+        - Roles
+      description: Adds a Permission to a Role
+      operationId: addRolePermission
+      security: []
+      parameters:
+        - in: path
+          name: role_name
+          description: Role name
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Permission Blueprint
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PermissionBlueprintDTO'
+      responses:
+        200:
+          description: Permission to a Role is added
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RoleDTO'
+        400:
+          description: Access type NONE not supported
```

Comment:
There is one more reason that this could return a 400: when the role_name is null or empty, so this could be: `Access type NONE not supported, or role name is null or empty`

Reply:
Done.